virttest/qemu_devices/qcontainer.py

```diff
@@ -1037,16 +1037,25 @@ def pflash_handler(firmware_name, cmd):
                 # To ignore the influence from backends
                 first_image = params.objects('images')[0]
                 img_params = params.object_params(first_image)
-                img_params["backing_chain"] = "yes"
+                img_params["backing_chain"] = "no"
                 path = storage.get_image_filename_filesytem(img_params,
                                                             current_data_dir)
                 img_info = json.loads(qemu_storage.QemuImg(img_params,
                                                            data_dir.DATA_DIR,
                                                            path).info(True, "json"))
-                img_name = os.path.basename(img_info[-1].get("filename"))
+
+                pflash_vars_name = (os.path.basename(img_info.get("filename"))
+                                    + "_VARS.fd")
                 # Force the image file to be saved under "images" directory
                 pflash_vars_path = os.path.join(data_dir.DATA_DIR, "images",
-                                                img_name) + "_VARS.fd"
+                                                pflash_vars_name)
+                # When image has backing files, treat it as a temporary image
+                if "backing-filename" in img_info:
+                    pflash_vars_path = os.path.join(data_dir.DATA_DIR, "images",
+                                                    "%s_%s" % (self.vmname,
+                                                               pflash_vars_name))
```

Comment:
How about using the same name pattern for the both scenarios?
`${self.vmname}_${filename}_VARS.fd`

Reply:
Yep, it should be no problem to use the same filename in all scenarios.
`vmname` + `filename` is the most suitable combination. Just not sure if it will affect the previous usage.
With this suggestion, the code can also be slightly simplified.

So yes, I agree with this suggestion but need the feature owner to do a sanity test.